app/lib/actions/katello/host/update_system_purpose.rb

```diff
@@ -0,0 +1,28 @@
+module Actions
+  module Katello
+    module Host
+      class UpdateSystemPurpose < Actions::EntryAction
+        def plan(host, service_level, purpose_role, purpose_usage)
+          if host.content_facet
+            host.subscription_facet.service_level = service_level
+            host.subscription_facet.purpose_role = purpose_role
+            host.subscription_facet.purpose_usage = purpose_usage
+
+            host.update_candlepin_associations
+            plan_self(:hostname => host.name)
+          else
+            fail _("Host %s has not been registered with subscription-manager.") % host.name
+          end
+        end
+
+        def humanized_name
+          if input.try(:[], :hostname).nil?
```

Comment:
I don't think I'm following this code. Can the hostname be nil?

Reply:
I removed this, I saw it used in other tasks, so just copied and pasted. After reading it does not make sense.